services/ossf-scorecard/ossf-scorecard.service.js

```diff
@@ -0,0 +1,59 @@
+import Joi from 'joi'
+import { BaseJsonService } from '../index.js'
+import { colorScale } from '../color-formatters.js'
+
+const schema = Joi.object({
+  score: Joi.required(),
+}).required()
+
+const ossfScorecardColorScale = colorScale(
+  [2, 5, 8, 10],
+  ['red', 'yellow', 'yellowgreen', 'green', 'brightgreen']
+)
+
+export default class OSSFScorecard extends BaseJsonService {
+  static category = 'analysis'
+
+  static route = { base: 'ossf-scorecard', pattern: ':host/:orgName/:repoName' }
+
+  static examples = [
+    {
+      title: 'OSSF-Scorecard Score',
+      namedParams: {
+        host: 'github.com',
```

Comment:
Is your intent to support integrating with server/self-hosted instances of the product(s) you'll support? I ask because we've a very large number of badges that are based on the big source code platforms, but we typically don't require our users to include the TLD for those platforms.

If the OSSF service does/will support any arbitrary git host then it obviously becomes necessary to incorporate more of the uri, although if that's the case then we'd need to rethink the route a bit to account for scenarios where people have opted to host their self-hosted instance at some non-root level path, e.g. my-awesome-github-enterprise-server.com/some-path/my-org/my-repo

Reply:
As of now, the primary audience for Scorecard badges is public repositories hosted on GitHub. Although the Scorecard team has plans of supporting non-GitHub VCS and non-public (say enterprise servers), we aren't there yet. So I don't want to overcomplicate the design here by trying to incorporate a generic uri. What if we update the uri to: `/github/:host/:repo`? If we plan to extend this in the future for other platforms, we can consider a different badge service. Would that work? @calebcartwright

Reply:
> If we plan to extend this in the future for other platforms, we can consider a different badge service. Would that work? @calebcartwright

No it would not. We don't want to be in the business of having duplicative routes/codes/etc. within Shields based on what a given provider may or may not do.

Let's just leave this as-is for now, and if/when you all make relevant changes upstream we can adjust the parameter in the route to support any optional extra paths